.pre-commit-config.yaml

```diff
@@ -1,14 +1,16 @@
 ---
 # See https://pre-commit.com for more information
+fail_fast: true
 # See https://pre-commit.com/hooks.html for more hooks
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v4.2.0
     hooks:
       - id: trailing-whitespace
         # Generated file
-        exclude: ^web/src/api/schema.d.ts|extra/.*$
+        exclude: ^web/src/api/schema\.d\.ts|extra/.*$
       - id: end-of-file-fixer
+        exclude: Cargo.nix
```

Comment:
note: Ideally, this would be fixed upstream, but for now this is fine.

Reply:
Upstream maintainers might not see it as something needing to be fixed (some people don't care about trailing new lines), so I think this is not only _for now_.

What would be nice if the hook could ignore generated files, eg: what we have marked as `linguist-generated` in `.gitattributes` so we didn't have to maintain two sets of rules.

Reply:
> What would be nice if the hook could ignore generated files, eg: what we have marked as linguist-generated in .gitattributes so we didn't have to maintain two sets of rules.

That would be really nice indeed.